src/test/integration/org/apache/cassandra/sidecar/routes/BaseTokenRangeIntegrationTest.java

```diff
@@ -0,0 +1,795 @@
+        return cassandraTestContext.configureAndStartCluster(
+        builder -> {
+            builder.withInstanceInitializer(initializer);
+            builder.withTokenSupplier(mdcTokenSupplier);
+            builder.withNodeIdTopology(networkTopology(totalNodeCount,
+                                                       (nodeId) -> nodeId % 2 != 0 ?
+                                                                   dcAndRack("datacenter1", "rack1") :
+                                                                   dcAndRack("datacenter2", "rack2")));
+        });
+    }
+
+    protected List<Range<BigInteger>> generateExpectedRanges()
+    {
+        CassandraIntegrationTest annotation = sidecarTestContext.cassandraTestContext().annotation;
+        int nodeCount = (annotation.nodesPerDc() + annotation.newNodesPerDc()) * annotation.numDcs();
+        return generateExpectedRanges(nodeCount);
+    }
+
+    protected List<Range<BigInteger>> generateExpectedRanges(int nodeCount)
+    {
+        CassandraIntegrationTest annotation = sidecarTestContext.cassandraTestContext().annotation;
+        TokenSupplier tokenSupplier = (annotation.numDcs() > 1) ?
+                                      MultiDcTokenSupplier.evenlyDistributedTokens(
+                                      annotation.nodesPerDc() + annotation.newNodesPerDc(),
+                                      annotation.numDcs(),
+                                      1) :
+                                      TokenSupplier.evenlyDistributedTokens(annotation.nodesPerDc() +
+                                                                            annotation.newNodesPerDc(),
+                                                                            1);
+
+        List<Range<BigInteger>> expectedRanges = new ArrayList<>();
+        BigInteger startToken = Partitioner.Murmur3.minToken;
+        BigInteger endToken = Partitioner.Murmur3.maxToken;
+        int node = 1;
+        BigInteger prevToken = new BigInteger(tokenSupplier.tokens(node++).stream().findFirst().get());
+        Range<BigInteger> firstRange = Range.openClosed(startToken, prevToken);
+        expectedRanges.add(firstRange);
+        while (node <= nodeCount)
+        {
+            BigInteger currentToken = new BigInteger(tokenSupplier.tokens(node).stream().findFirst().get());
+            expectedRanges.add(Range.openClosed(prevToken, currentToken));
+            prevToken = currentToken;
+            node++;
+        }
+        expectedRanges.add(Range.openClosed(prevToken, endToken));
+        return expectedRanges;
+    }
+
+    protected Set<String> instancesFromReplicaSet(List<TokenRangeReplicasResponse.ReplicaInfo> replicas)
+    {
+        return replicas.stream()
+                       .map(r -> r.replicasByDatacenter().values())
+                       .flatMap(Collection::stream)
+                       .flatMap(Collection::stream)
+                       .collect(Collectors.toSet());
```

Comment:
nit: one less line if you do this 
```suggestion
        return replicas.stream()
                       .flatMap(r -> r.replicasByDatacenter().values().stream())
                       .flatMap(Collection::stream)
                       .collect(Collectors.toSet());
```